### ISSUE TITLE
Unhandled attempt to divide by zero exception

### ISSUE THREAD

**Issue by Somfic** [Speykious/SeeShark#50]
Hi, 

Getting a `System.DivideByZeroException` at 

```
at SeeShark.Device.VideoDevice.TryGetFrame(Frame& frame)
at SeeShark.Device.VideoDevice.DecodeLoop()
``` 

which most likely is when the device returns a framerate of 0 (?). See https://github.com/Speykious/SeeShark/blob/cc4aa3b69053f08c92b77528ab95d326c91d7f72/SeeShark/Device/VideoDevice.cs#L88C12-L88C86.

```cs
Thread.Sleep(1000 * decoder.Framerate.den / (decoder.Framerate.num * 4));
```

**Comment by Somfic** [comment]
sick, thanks! can’t test right now as I am not near my machine but i’ll close this issue once i confirm it’s working :)

**Comment by Speykious** [comment]
If you're on Linux I'm sure it'll work, but I'm just in the process of rewriting the library, so it won't mean much if it works at this stage. xD
I removed almost everything, including `VideoDevice`, the asynchronous API, the decoders... Because I decided that it would be best to start rewriting the abstraction once I have multiple APIs in place and working correctly.

I'd say I'm already as good as done for V4l2 on Linux, after just a few days of coding this, which is nice.
But I still need to deal with all the other APIs: X11, potentially Wayland, then AVFoundation on MacOS, and DirectShow and Gdi on Windows.
